Meetup 27 Resources (#206)
w3c/wot-cg: Meetups/27/2025-06-05-minutes.md

### :bust_in_silhouette: Participants

<!-- This list will copied over from the meeting tool -->
- Ege Korkan
- Cristiano Aguzzi
- Thomas Wehr
- Jan Hofmann
- Kunihiko Toumura
- Thomas Jaeckle
- Zhou Gui
- Andreas Harth
- Jonathan Wise
- Tatiana Mangels
- York Schmidtke
- Daniel Peintner

:writing_hand: **Scribe:** Cris

----

### Introduction

Ege: Welcome Everyone! we are at Meetup 27th! All slide are public
Ege: you can find us in different places on the web

### News

Ege: we do not have new meetups planned in the future but you can join the office hours to talk with us.

Ege: we have two new tutorial videos! please visit our youtube channel

Ege: there are more underway, if you are interested we are meeting every friday. You can help us with new ideas for tutorials

Ege: This year TPAC is planned in Kyoto, Japan, for 10-14 November. We are going to join as CG please let us know if you want to attend.

Ege: join the CG to keep in touch or to propose a new meetup!

### Meetup Presentation

Ege: Today we have Thomas Wehr, he is a PhD at the Chair for Technical Information Systems. In the past his department have participated in different WoT projects.

> Thomas: We developed a tool to visual create WoT consumer. We wanted to create a programming environment that should portable and usable as many devices as possible. Moreover, it should be very accessible (that's why we opted for a visual programming interface). We use the Web of Things to make interoperable with many type of devices.

Reviewer: ```suggestion
Thomas: We developed a tool to visual create WoT consumer. We wanted to create a programming environment that should be portable and usable on as many devices as possible. Moreover, it should be very accessible (that's why we opted for a visual programming interface). We use the Web of Things to make it interoperable with many type of devices.
```

> Thomas: visual programming in general refers to a programming environment that you arrange graphical elements to create programs (workflow, flowchart, and block based). We opted for a block based approach and we called Blast.

Reviewer: ```suggestion
Thomas: "Visual programming" in general refers to a programming environment in which you arrange graphical elements to create programs (typically workflow-, flowchart-, and block-based). We opted for a block-based approach which we call _**Blast**_.
```

> Thomas: we use Blockly from Google and we adapted it to Web of Things concepts
> 
> Thomas: there are many pre-defined blocks in Blockly: string operations, mathematical operations, speech to text and vice versa, http requests, etc.

Reviewer: ```suggestion
Thomas: We used Blockly from Google, and we adapted it to Web of Things concepts.

Thomas: There are many pre-defined blocks in Blockly: string operations, mathematical operations, speech-to-text and vice versa, HTTP requests, etc.
```

Thomas: Blast has the concept of state. The main feature is to consume a Thing Description and change the device without changing any code. Blast can list only the properties available in the TD and also can validate the input/output of a block using TD data schema.

> Thomas: Blast is built on top of node-wot. It worked out-of-the-box for us, but we needed a protocol binding for Bluetooth devices. The second extension that we had to do is to support "scale" and "default". The "scale" was useful for a particular device called RuuviTag and the default for a BLE LED controller. The default keyword was useful to write concise data payloads.

Reviewer: ```suggestion
Thomas: Blast is built on top of `node-wot`. It worked out-of-the-box for us, but we needed a protocol binding for Bluetooth devices. The second extension that we had to make was to support "scale" and "default". The "scale" was useful for a particular device called RuuviTag and the default for a BLE LED controller. The default keyword was useful to write concise data payloads.
```

> Thomas: let's see Blast in the action.

Reviewer: ```suggestion
Thomas: Let's see Blast in action.
```

> Shows Blast demo
> Thomas program a Stream deck to connect with a Lamp

> Thomas: to concluded that was Blast a visual programming and execution environment for WoT Consumers. It can consume TDs to dynamically to add new devices.

Reviewer: ```suggestion
Thomas: To conclude, that was Blast, a visual programming and execution environment for WoT Consumers. It can consume TDs to dynamically to add new devices.
```

> Thomas: we used node-wot but we extend it with a new binding and a better codec for octet streams.

Reviewer: ```suggestion
Thomas: We used `node-wot`, but we extended it with a new binding and a better codec for octet streams.
```

### Discussion

> Ege: thank you for using our open source tools!
> 
> Ege: you showed a list of devices, is pre-configured or dynamically loaded via some sort of discovery method.

Reviewer: ```suggestion
Ege: Thank you for using our open source tools!

Ege: You showed a list of devices. Is that pre-configured, or dynamically loaded via some sort of discovery method?
```

> Thomas: is predefined.
> 
> Ege: could you use BLE to discover Things and get their TDs from a catalog?

Reviewer: ```suggestion
Thomas: It is predefined.

Ege: Could you use BLE to discover Things and get their TDs from a catalog?
```

> Thomas: yes, that would be possible.
> 
> Ege: this is similar approach to node-red, why did you choose this approach?

Reviewer: ```suggestion
Thomas: Yes, that would be possible.

Ege: This is similar approach to `node-red`. Why did you choose this approach?
```

> Thomas: Node-red is data oriented, here we have more flexibility because is a full programming language.
> 
> Cris: can you consume Thing Models? maybe they can be even used as a skeleton for program arguments

Reviewer: ```suggestion
Thomas: `Node-red` is data oriented. Here we have more flexibility, because it is a full programming language.

Cris: Can you consume Thing Models? Maybe they can even be used as a skeleton for program arguments?
```

Thomas: It is possible, we are using it for some BLE devices.

> Daniel: thank you for creating a new protocol binding for node-wot, we should find a way to list it somewhere. Do you have an online version of the BLAST environment?
> 
> Thomas: Yes we have it hosted online: <https://wintechis.github.io/blast/>

Reviewer: ```suggestion
Daniel: Thank you for creating a new protocol binding for `node-wot`. We should find a way to list it somewhere. Do you have an online version of the BLAST environment?

Thomas: Yes, we have it hosted online: <https://wintechis.github.io/blast/>
```

> Ege: do you have some spec document about the binding you developed.
> 
> Thomas: we don't have a lot of documatation for that yet.

Reviewer: ```suggestion
Ege: Do you have some spec document about the binding you developed?

Thomas: We don't have a lot of documentation for it yet.
```

> Ege: soon in the WG we'll have a way to create a document about a protocol binding. Are you interested in this activity?
> 
> Thomas: sure.

Reviewer: ```suggestion
Ege: Soon, in the WG, we'll have a way to create a document about a protocol binding. Are you interested in this activity?

Thomas: Sure.
```

> Ege: what is the difference between using BLAST in the browser or locally?
> 
> Thomas: There is not great interest to support BLE in browsers.

Reviewer: ```suggestion
Ege: What is the difference between using BLAST in the browser or locally?

Thomas: There is not great interest to support BLE in browsers.
```

> Ege: do you know why?
> 
> Thomas: not really...
> 
> Cris: where is the binding hosted?
> 
> Thomas: it is in our repository but the concrete implementation is split over two different "modules" one for node and the other for browser.

Reviewer: ```suggestion
Ege: Do you know why?

Thomas: Not really...

Cris: Where is the binding hosted?

Thomas: It is in our repository, but the concrete implementation is split over two different "modules"; one for `node` and the other for browser.
```

> Cris: are you planning to move to deno?
> 
> Thomas: not yet, but we are trying to port it there.
> 
> Cris: we would be happy to know more if node-wot works out of the box there.

Reviewer: ```suggestion
Cris: Are you planning to move to deno?

Thomas: Not yet, but we are trying to port it there.

Cris: We would be happy to know more if `node-wot` works out of the box there.
```

> Ege: is there anything that would help you if it were in the standard?
> 
> Thomas: I not really sure
> 
> Ege: when you generate the UI for the inputs of the TD what is the logic are you using?
> 
> Thomas: steam deck block are statically generated

Reviewer: ```suggestion
Ege: Is there anything that would help you if it were in the standard?

Thomas: I am not really sure.

Ege: When you generate the UI for the inputs of the TD, what is the logic you are using?

Thomas: Steam deck blocks are statically generated.
```
